platform/src/ToolsManager.js

```diff
@@ -31,8 +31,26 @@ class ToolManager {
                 
                 let toolUrl = new Object();
                 toolUrl.id = ""; // Populate when tool is fetched
-                toolUrl.url = url;
-
+                if (this.isValidUrl(url)){
+                    toolUrl.url = url;
+                }
+                else{
+                    let url_tail = url.split('/')[1];
+                    let url_port = this.getPort(url);
+                    if (url_port != null){
+                        let path = this.fetchPathByPort(url_port);
+
+                        if(path != null){
+                            let base_url = utility.getBaseURL();
+                            path = path.endsWith('/') ? path : path + '/';
+                            toolUrl.url = base_url + path + url_tail;
+                        }
```

Comment:
There should be an error notification here.

Reply:
Added.